### PR TITLE
feat: expose render progress as argument

### PR DESCRIPTION
**Pull request by justusmattern27**

### CODE REVIEW

File: packages/docs/docs/api-reference/renderVideo.mdx [hunk 1]
```diff
@@ -13,15 +13,16 @@ process. It uses a headless browser to achieve this.
 ```tsx
 import {renderVideo} from '@revideo/renderer';
 
-function logProgress(id: number, progress: number) {
+function printProgress(id: number, progress: number) {
```
[the remark this answers is not in the document]
Comment (justusmattern27) [reply packages/docs/docs/api-reference/renderVideo.mdx:16]: we set logProgress: true as an argument in RenderVideoSettings - I just wanted to make sure that people are not confused about these two things having the same name

File: packages/create/template-2d-ts/src/render.ts [hunk 2]
```diff
@@ -1,40 +1,16 @@
...
-  const interval = setInterval(() => {
-    printProgress();
-  }, 1000);
-
   // This is the main function that renders the video
   const file = await renderVideo(
     './vite.config.ts',
     {fill: 'orange'},
-    trackProgress,
+    () => {},
```
[the remark this answers is not in the document]
Comment (justusmattern27) [reply packages/create/template-2d-ts/src/render.ts:10]: personally I would be pretty confused about passing undefined as an argument. It is probably best anyway to use the object as function input, then we don't have that problem anymore

File: packages/renderer/server/render-video.ts [hunk 3]
```diff
@@ -94,7 +95,27 @@ async function renderVideoOnPage(
   server: ViteDevServer,
   url: string,
   progressCallback?: (worker: number, progress: number) => void,
+  logProgress?: boolean,
 ) {
+  const progressTracker = new Map<number, number>();
+
+  function printProgress() {
```
[the remark this answers is not in the document]
Comment (justusmattern27) [reply packages/renderer/server/render-video.ts:101]: done

File: packages/create/template-2d-ts/src/render.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (hkonsti) [reply packages/create/template-2d-ts/src/render.ts:10]: fair 👍

**Review by hkonsti — APPROVED** [review]
looks good 👍

**Pull request merged**